### DISCUSSION TITLE
Creating own variable in Shell script

### DISCUSSION THREAD

**Discussion by ais-one** [community/community#25459] [Actions]
<p>I am running a shell script in an action and has the contents below</p>
<pre><code class="lang-auto">MY_APP='my-app'
BUILD_TS=`date +"%Y%m%d%H%M"`

docker build -t gcr.io/some-project/$MY_APP:$BUILD_TS
</code></pre>
<p>Will it be affected by the following change below?</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/" target="_blank" rel="noopener nofollow ugc">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/03/blog-card.png?fit=5001%2C2626" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/" target="_blank" rel="noopener nofollow ugc">GitHub Actions: Deprecating set-env and add-path commands - GitHub Changelog</a></h3>

<p>GitHub Actions: Deprecating set-env and add-path commands</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>If so other than passing in arguments, what can be done? especially since I would really like to have $BUILD_TS</p>
<p>The shell script…</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/ais-one/vue-crud-x/blob/96e7850cdd12ec3272f99494b4bd92e72285fed2/example-app/deploy.sh" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/ais-one/vue-crud-x/blob/96e7850cdd12ec3272f99494b4bd92e72285fed2/example-app/deploy.sh" target="_blank" rel="noopener nofollow ugc">ais-one/vue-crud-x/blob/96e7850cdd12ec3272f99494b4bd92e72285fed2/example-app/deploy.sh</a></h4>
<pre><code class="lang-sh">#!/bin/bash
# $0. The name of script itself.
# $$ Process id of current shell.
# $* Values of all the arguments. ...
# $# Total number of arguments passed to script.
# $@ Values of all the arguments.
# $? Exit status id of last command.

echo $CI

if [ ! $1 ]; then # environment eg. uat
  echo "Missing project environment. Set at package.json. Press any key to continue..."
  if [ "$CI" != "true" ]; then
    read
  fi
  exit
fi

if [ "$1" = "development" ]; then
  echo "Cannot deploy using development environment. Press any key to continue..."
</code></pre>

  This file has been truncated. <a href="https://github.com/ais-one/vue-crud-x/blob/96e7850cdd12ec3272f99494b4bd92e72285fed2/example-app/deploy.sh" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by Simran-B** [comment]
<p>What you define there are variables, not environment variables. You also don’t echo <code>::set-env</code> or <code>::add-path</code>, which are the two workflow commands that are deprecated.</p>

**Comment by ais-one** [comment]
<p>ok thanks, then it could be something else that was generating those warnings…</p>
<p>I think I might have used something like $GCP_PROJECT_ID which is also set it github secrets… which caused this?</p>
<p>I added link to the shell script file at the start of the topic</p>

**Comment by Simran-B** [comment, marked as answer]
<p>Many actions including official GitHub actions use outdated dependencies.</p>
<p>Version 1.2.6 of <code>@actions/core</code> implements the new way of setting environment variables using a file. Action authors will have to update their dependencies.</p>
<p>Any action which uses an older version may cause a depreciation warning in the annotations of workflow runs.</p>
